pkg/stanza/fileconsumer/config.go

```diff
@@ -165,6 +178,13 @@ func (c Config) validate() error {
 		return fmt.Errorf("`fingerprint_size` must be at least %d bytes", MinFingerprintSize)
 	}
 
+	if c.DeleteAfterRead && c.StartAt == "end" {
+		if !featuregate.GetRegistry().IsEnabled(allowFileDeletion) {
+			return fmt.Errorf("`delete_after_read` requires `%s` feature gate", allowFileDeletion)
+		}
+		return fmt.Errorf("`delete_after_read` cannot be used with `start_at: end`")
+	}
```

Comment:
Looks like the featuregate here will not be checked in the case that `delete_after_read` is true, but `start_at` is "beginning", allowing the feature to be used without the feature gate.

Reply:
Good catch. I decided to separate the feature gate checking into 1) whether the value _can_ be set 2) whether the set value is valid.